lib/broadway.ex

```diff
@@ -291,6 +291,33 @@ defmodule Broadway do
   is your responsibility to configure a dead-letter queue as target
   for those messages.
 
+  ## Producer concurrency
+
+  For efficiency, you should generally limit the amount of internal queueing.
```

Comment:
Thanks for the PR!

We probably need to clarify here this behaviour in relation to poll producers. You may actually prefer internal queuing if cost of fetching the messages is too high. With RabbitMQ that's a very easy equation to solve due to the prefetch_count and the general connection pattern, but for AmazonSQS you may prefer to have it sitting idle on your inbox rather than having to wait for a full round trip before you can finally do more work.

Reply:
Gotcha - I'll update accordingly. Thanks for explaining.

Reply:
Updated!